src/lib.rs

```diff
@@ -71,6 +71,44 @@ impl PoseidonHasher {
     pub fn hash_no_pad(x: Vec<GoldilocksField>) -> Vec<u8> {
         PoseidonHash::hash_no_pad(&x).to_bytes()
     }
+
+    const EXPECTED_STORAGE_PREIMAGE_LEN: usize = 4 + 32 + 32 + 16;
+    
+    // This function should only be used to compute the quantus storage key for Transfer Proofs
+    // It breaks up the bytes input in a specific way that mimics how our zk-circuit does it
+    pub fn hash_storage(x: &[u8]) -> [u8; 32] {
+        // TODO: probably shouldn't be an assert
+        assert_eq!(x.len(), Self::EXPECTED_STORAGE_PREIMAGE_LEN);
```

Comment:
Yeah, I think this should definitely return an error somehow. Ideally it should take in a fixed input array to make it a compile-time error, but I don't think that's possible with the current trait bounds